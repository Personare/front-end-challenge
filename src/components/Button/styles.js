import styled from "styled-components";

const ButtonContainer = styled.button`
  background: #2F4A94;
  border-radius: 8px;
  box-shadow: 0 10px 10px 0 rgba(0, 0, 0, 0.2);
  font-size: 1.5rem;
  color: #FFF;

  div {
    width: 100%;
    height: 100%;
    padding: 10px 25px;
    cursor: pointer;
  }
`;

export default ButtonContainer;
