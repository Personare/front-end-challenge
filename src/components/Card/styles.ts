import styled, { css } from "styled-components";

interface ImageProps {
  isMinimal?: boolean;
}
export const Image = styled.img<ImageProps>`
  ${({ isMinimal }) => css`
    height: ${isMinimal ? "150px" : "300px"};
    width: ${isMinimal ? "79px" : "auto"};
    margin-bottom: ${isMinimal ? "10px" : "auto"};
  `}
`;

export const Container = styled.div``;
