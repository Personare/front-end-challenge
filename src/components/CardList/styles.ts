import styled from "styled-components";

export const List = styled.div`
  align-items: center;
  display: grid;
  gap: 20px;
  width: 100%;

  flex-direction: column;
  grid-template-columns: repeat(auto-fill, minmax(162px, 1fr));

  img {
    display: block;
    height: auto;
    width: 100%;
  }
`;
