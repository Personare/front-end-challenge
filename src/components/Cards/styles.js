import styled from "styled-components";

const CardsContainer = styled.section`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: flex-start;
  max-width: 780px;
  margin-top: 30px;

  & > img {
    height: 230px;
    width: 109.26px;
    margin: 15px 10px;
    cursor: ${({ gameOn }) => gameOn ? "pointer": "default"};
  }
`;

const FlipCard = styled.div`
  background-color: transparent;
  width: 109.26px;
  height: 230px;
  margin: 15px 10px;
  perspective: 1000px;

  img {
    height: 100%;
    width: 100%;
    cursor: ${({ gameOn }) => gameOn ? "pointer": "default"};
  }

  :hover > div {
    transform: rotateY(180deg);
  }

  & > div {
    transform: ${({ gameOn }) => gameOn ? "rotateY(180deg)": "none"};
  }
`;

const FlipCardInner = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  text-align: center;
  transition: all 0.6s;
  transform-style: preserve-3d;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
`;


const CardFront = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
`;


const CardBack = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  transform: rotateY(180deg);
`;


export {
  CardsContainer,
  FlipCard,
  FlipCardInner,
  CardFront,
  CardBack
};
