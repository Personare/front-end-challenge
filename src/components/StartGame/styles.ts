import styled from "styled-components";

export const Container = styled.div`
  display: grid;
  grid-template-areas: "image action";
  grid-template-columns: 50% minmax(auto, 50%);
  grid-template-rows: 100vh;

  @media (max-width: 768px) {
    grid-template-areas: "action" "image";
    grid-template-columns: 1fr;
    grid-template-rows: minmax(50vh, 1fr) 1fr;
  }
`;

export const ImageContainer = styled.div<{ src: string }>`
  background: ${({ src }) => `url(${src})`} no-repeat;
  background-size: cover;
  grid-area: image;
`;

export const Image = styled.img`
  max-width: 100%;
  display: block;
`;

export const ButtonContainer = styled.div`
  grid-area: action;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;

  @media (max-width: 768px) {
    margin: 40px 0;
  }
`;

export const Title = styled.h1``;

export const Subtitle = styled.p`
  margin: 15px 0 30px 0;
  font-size: 12px;
`;

export const PlayButton = styled.button`
  padding: 15px 60px;
  border-radius: 6px;
  outline: none;
  border: none;
  background-color: #2e3f61;
  color: #fff;
  font-weight: 700;
  transition: background-color 300ms ease;

  &:hover {
    cursor: pointer;
    background-color: rgba(46, 63, 97, 0.8);
  }
`;
