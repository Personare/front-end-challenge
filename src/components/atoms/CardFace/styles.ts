import styled from 'styled-components';

type CardFaceProps = {
  rotate: string;
};

export const CardFace = styled.div<CardFaceProps>`
  backface-visibility: hidden;
  height: inherit;
  overflow: hidden;
  position: absolute;
  transform: perspective(200px) rotateY(${({ rotate }) => rotate});
  transition: transform 0.6s linear;
`;

export const CardImage = styled.img`
  height: 100%;
  object-fit: cover;
  object-position: center;
  width: 100%;
`;
