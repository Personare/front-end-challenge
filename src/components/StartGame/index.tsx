import * as S from "./styles";

import BackgroundImage from "../../assets/images/background-image.jpg";

import { IStartGame } from "../../config/interfaces";

const StartGame: React.FC<IStartGame> = ({ onClick }) => {
  return (
    <S.Container data-cy="start">
      <S.ImageContainer src={BackgroundImage} />

      <S.ButtonContainer>
        <S.Title>Portal Tarot</S.Title>
        <S.Subtitle>
          Clique no botão abaixo para ver a previsão do dia
        </S.Subtitle>

        <S.PlayButton type="button" onClick={onClick} data-cy="action-button">
          Iniciar
        </S.PlayButton>
      </S.ButtonContainer>
    </S.Container>
  );
};

export default StartGame;
