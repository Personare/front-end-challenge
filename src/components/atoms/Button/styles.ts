import styled from 'styled-components';
import { colors } from 'styles/variables';

export const Button = styled.button`
  background: ${colors.pink};
  border: 0;
  border-bottom: 4px solid rgba(0, 0, 0, 0.3);
  border-radius: 8px;
  box-shadow: 0 4px 8px ${colors.gray};
  color: ${colors.white};
  cursor: pointer;
  font-family: 'Open Sans', sans-serif;
  font-size: 18px;
  font-weight: 600;
  letter-spacing: 0.05em;
  padding: 12px 24px;

  :hover {
    opacity: 0.9;
  }

  &[disabled] {
    cursor: initial;
    opacity: 0.5;
  }
`;
