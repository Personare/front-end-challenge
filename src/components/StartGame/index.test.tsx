import { shallow } from "enzyme";
import StartGame from ".";
import { PlayButton } from "./styles";

describe("StartGame", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<StartGame />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should change the component when clicking on button", () => {
    const onClickFn = jest.fn();
    const wrapper = shallow(<StartGame onClick={onClickFn} />);

    expect(onClickFn).not.toHaveBeenCalled();

    wrapper.find(PlayButton).simulate("click");

    expect(onClickFn).toHaveBeenCalledTimes(1);
  });
});
