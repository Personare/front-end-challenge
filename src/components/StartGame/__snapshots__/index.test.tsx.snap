// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartGame should match snapshot 1`] = `
<styled.div
  data-cy="start"
>
  <styled.div
    src="background-image.jpg"
  />
  <styled.div>
    <styled.h1>
      Portal Tarot
    </styled.h1>
    <styled.p>
      Clique no botão abaixo para ver a previsão do dia
    </styled.p>
    <styled.button
      data-cy="action-button"
      type="button"
    >
      Iniciar
    </styled.button>
  </styled.div>
</styled.div>
`;
